### PR TITLE
Bratseth/float tensor

### PR DESCRIPTION
**Pull request by bratseth**
Create, encode and decode, but not compute, with indexed bound float tensors without any conversion to double.

@lesters please review

### CODE REVIEW

File: vespajlib/src/main/java/com/yahoo/tensor/IndexedTensor.java [hunk 1]
```diff
@@ -245,10 +253,16 @@ public static Builder of(TensorType type, DimensionSizes sizes) {
                                                        " but cannot be larger than " + size.get() + " in " + type);
             }
 
-            return new BoundBuilder(type, sizes);
+            if (type.valueType() == TensorType.Value.FLOAT)
+                return new IndexedFloatTensor.BoundFloatBuilder(type, sizes);
+            else if (type.valueType() == TensorType.Value.FLOAT)
```
Comment (lesters) [inline vespajlib/src/main/java/com/yahoo/tensor/IndexedTensor.java]: Same condition in if, should probably be ` == TensorType.Value.DOUBLE` here.

**Review by lesters — CHANGES_REQUESTED** [review]
Overall, LGTM. I guess unit tests are incoming when we get to computation of tensors with floats. Please check logic mentioned in comment above.

File: vespajlib/src/main/java/com/yahoo/tensor/IndexedTensor.java (hunk 1 above)
Comment (bratseth) [reply vespajlib/src/main/java/com/yahoo/tensor/IndexedTensor.java]: Thanks! Functionally correct, but logically wrong ...

**Comment by bratseth** [comment]
Added an explicit test of this too - merging.

**Pull request merged**